src/state/connection/index.ts

```diff
@@ -1,52 +1,60 @@
-import { create } from 'zustand';
 import { PenumbraRequestFailure, PenumbraState, PenumbraManifest } from '@penumbra-zone/client';
 import { penumbra, PRAX_ORIGIN } from '@/utils/penumbra';
+import { makeAutoObservable } from 'mobx';
 
-export interface ConnectionState {
-  connected: boolean;
+class ConnectionStateStore {
+  connected = false;
   manifest: PenumbraManifest | undefined;
-  reconnect: () => Promise<void>;
-  connect: () => Promise<void>;
-  disconnect: () => Promise<void>;
-  /** Call this function in the global `useEffect` hook to subscribe to provider connection change */
-  setup: () => void;
-}
 
-export const useConnectionState = create<ConnectionState>()((set, get) => ({
-  connected: false,
-  manifest: undefined,
+  constructor() {
+    makeAutoObservable(this);
+
+    if (typeof window !== 'undefined') {
+      this.setup();
+    }
```

Comment:
A substitute for on mount useEffect() hook. Downside is it needs a window guard to ensure it only runs in the client.